src/data_provider/src/network/networkLinuxWrapper.h

```diff
@@ -222,12 +225,43 @@ class NetworkLinuxInterface final : public INetworkInterfaceWrapper
 
 public:
     explicit NetworkLinuxInterface(ifaddrs* addrs)
-    : m_interfaceAddress(addrs)
+    : m_interfaceAddress{ addrs }
+    , m_gateway{"unknown"}
+    , m_metrics{"unknown"}
     { 
         if (!addrs)
         {
             throw std::runtime_error { "Nullptr instances of network interface" };
         }
+        else
+        {
+            auto fileData { Utils::getFileContent(std::string(WM_SYS_NET_DIR) + "route") };
+            const auto ifName { this->name() };
+            if (!fileData.empty())
+            {
+                auto lines { Utils::split(fileData, '\n') };
+                for (auto& line : lines)
+                {
+                    line = Utils::rightTrim(line);
+                    Utils::replaceAll(line, "\t", " ");
+                    Utils::replaceAll(line, "  ", " ");
+                    const auto fields { Utils::split(line, ' ') };
+
+                    if (GatewayFileFields::Size == fields.size() &&
+                        fields.at(GatewayFileFields::Iface).compare(ifName) == 0)
+                    {
+                        const auto address { static_cast<uint32_t>(std::stoi(fields.at(GatewayFileFields::Gateway), 0, 16)) };
+                        m_metrics = fields.at(GatewayFileFields::Metric);
+
+                        if (address)
+                        {
+                            m_gateway = std::string(inet_ntoa({ address }));
```

Comment:
If I'm not wrong, gateway could be a list of values and here `m_gateway` will always have only 1 value. Is this the expected behavior?

Reply:
In linux based on old syscollector implementation, can only return 1 value.